bump-lockfile job for s390x

LakshmiRavichandran1 opened the pull request:
add steps to the bump-lockfile job for s390x

dustymabe commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -181,6 +188,28 @@ EOF
                 shwrap("tar -cf - tmp/kola-testiso-uefi/ | xz -c9 > ${env.WORKSPACE}/kola-testiso-uefi.tar.xz")
                 archiveArtifacts allowEmptyArchive: true, artifacts: 'kola-testiso*.tar.xz'
             }
+        }, s390x: {
```
we should maybe move the x86_64 section to the top or the bottom so the aarch64 and s390x sections can be next to one another. If you move it to the top maybe do it in a separate commit prep commit.

LakshmiRavichandran1 replied:
addressed in the recent commit.

dustymabe reviewed (commented):
Thanks for working on this! I need to add the s390x builder info into the cluster, but then we should be able to start testing this soon.

A few other comments.

LakshmiRavichandran1 commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -181,6 +188,28 @@ EOF
+        post_commands:
+        - cosa kola run --rerun --basic-qemu-scenarios
+        - rm -f builds/builds.json # https://github.com/coreos/coreos-assembler/issues/2317
+        delay_meta_merge: false
```
addressed in the recent commit.

LakshmiRavichandran1 commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -74,6 +74,13 @@ try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTE
+            def appendFlags = "--git-ref=${params.STREAM}"
+            appendFlags += " --git-url=https://github.com/${repo}"
+            appendFlags += " --returnFiles=src/config/manifest-lock.s390x.json"
+            gp.gangplankArchWrapper([cmd:                   "cosa fetch --update-lockfile --dry-run",
```
addressed in the recent commit.

LakshmiRavichandran1 commented:
@dustymabe could you please review the recent commit which addresses your comments?
I will wait for the s390x builder info to be added to the cluster as well.

dustymabe commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -137,6 +144,27 @@ delay_meta_merge: false
+    - rm -f builds/builds.json # https://github.com/coreos/coreos-assembler/issues/2317
+delay_meta_merge: false
+EOF
+               """)
+        gp.gangplankArchWrapper([spec: "spec.spec", arch: "s390x"])
```
```suggestion
                   """)
            gp.gangplankArchWrapper([spec: "spec.spec", arch: "s390x"])
```

spacing

LakshmiRavichandran1 replied:
addressed in commit #00c50f1

LakshmiRavichandran1 commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -137,6 +144,27 @@ delay_meta_merge: false
             gp.gangplankArchWrapper([spec: "spec.spec", arch: "aarch64"])
+        }, s390x: {
+        shwrap("""
+           cat <<'EOF' > spec.spec
```
addressed in commit #00c50f1

LakshmiRavichandran1 commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -74,6 +74,13 @@ try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTE
+            def appendFlags = "--git-ref=${params.STREAM}"
+            appendFlags += " --git-url=https://github.com/${repo}"
+            appendFlags += " --returnFiles=src/config/manifest-lock.s390x.json"
+            gp.gangplankArchWrapper([cmd:"cosa fetch --update-lockfile --dry-run",
```
addressed in commit #00c50f1

dustymabe reviewed (approved):
LGTM

dustymabe reviewed (changes requested):
OK I took another look at this. Some how there are commits missing from your PR branch here that are in the main branch. For example this whole block:

https://github.com/coreos/fedora-coreos-pipeline/blob/feb01eb32571518faeadd9e617107f77124e7aba/jobs/bump-lockfile.Jenkinsfile#L147-L160

Rebase your changes on top of latest `main` and then copy and paste to make the s390x section match the aarch64 section again. Also let's squash the commits down into a single commit.

LakshmiRavichandran1 commented:
2055ac9331a947f7d51c8f2fb2461392f59398ff has the squashed changes.

LakshmiRavichandran1 commented on jobs/bump-lockfile.Jenkinsfile:
```diff
@@ -162,6 +169,27 @@ delay_meta_merge: false
+  post_commands:
+    - cosa kola run --rerun --basic-qemu-scenarios
+    - rm -f builds/builds.json # https://github.com/coreos/coreos-assembler/issues/2317
+delay_meta_merge: false
```
I have made it equal now by removing 'build_artifacts' and adding 'prep_commands'.

LakshmiRavichandran1 replied:
> `kola testiso -SP --qemu-native-4k --scenarios iso-install --output-dir tmp/kola-metal4k`
there are currently no valid testing scenarios for metal4k images in case of s390x, so it is staged currently.
with this commit, the post commands are made equal to match aarch64

LakshmiRavichandran1 replied:
yes, it is needed to build the metal4k image as it feeds to build the live image which otherwise fails and live build is a needed artifact.

dustymabe reviewed (approved):
LGTM - thanks for working on this!
